src/audio/kpb.c

```diff
@@ -56,7 +56,10 @@ struct comp_data {
 	bool is_internal_buffer_full;
 	size_t buffered_data;
 	struct dd draining_task_data;
-	size_t buffer_size;
+	size_t kpb_buffer_size;
+	size_t host_buffer_size;
+	size_t period_size;
+	struct comp_dev *dev;
```

Comment:
Why do we need it? Every generic component API function operates on dev from which you can obtain comp_data, not other way around.

Reply:
@tlauda it is stored in component data so functions like buffering or draining can use it to finish requested reset. The reset procedure in KPB is a bit more complicated than it is in other components.

Reply:
I see it only used in `kpb_buffer_data` just to pass it to `kpb_reset`. You have function `kpb_copy`, which takes comp_dev and from comp_dev you are getting comp_data, which is passed to `kpb_buffer_data`, which needs to pass comp_dev retrieved from comp_data. It doesn't seem right to me.